### PR TITLE
Code review for the retry function

### PR DESCRIPTION
**Pull request by MaksGovor**

### CODE REVIEW

File: lib/retry.js [hunk 1]
```diff
@@ -4,37 +4,67 @@ const sleep = msec => new Promise(resolve => {
...
 });
 
-const promisify = fn => (...args) => (
-  new Promise((resolve, reject) => (
-    fn(...args, (err, data) => (
-      err ? reject(err) : resolve(data)
-    ))
-  ))
-);
+const promisify = (fn, args) => new Promise((resolve, reject) => {
```
Comment (tedi4t) [inline lib/retry.js]: Maybe make a separate file 'utils' and save promisify, deepClone etc. there?

File: lib/retry.js [hunk 2]
```diff
@@ -4,37 +4,67 @@ const sleep = msec => new Promise(resolve => {
...
+        multiplier,
+      });
+      if (presentCb && !result) return;
     }
   }
-  return res;
 };
 
-module.exports = { retry, sleep };
+module.exports = { retry };
```
Comment (tedi4t) [inline lib/retry.js]: maybe change it for `module.exports = retry;` because we exports only one item?

File: lib/retry.js [hunk 3]
```diff
@@ -4,37 +4,67 @@ const sleep = msec => new Promise(resolve => {
...
+  } else if (Array.isArray(obj)) {
+    const result = [];
+    for (const data of obj) {
+      result.push(deepClone(data));
+    }
+    return result;
+  } else {
+    const result = {};
+    const keys = Object.keys(obj);
+    for (const key of keys) {
```
Comment (tedi4t) [inline lib/retry.js]: Maybe use `for(key in obj)` instead?

File: lib/retry.js [hunk 4]
```diff
@@ -4,37 +4,67 @@ const sleep = msec => new Promise(resolve => {
...
 
-const promisify = fn => (...args) => (
-  new Promise((resolve, reject) => (
-    fn(...args, (err, data) => (
-      err ? reject(err) : resolve(data)
-    ))
-  ))
-);
+const promisify = (fn, args) => new Promise((resolve, reject) => {
+  if (!args) args = [];
```
Comment (Anastasia-Tre) [inline lib/retry.js]:
```suggestion
const promisify = (fn, args = []) => new Promise((resolve, reject) => {
```
Maybe better to use default argument

File: lib/retry.js (hunk 3 above)
Comment (MaksGovor) [reply lib/retry.js]: I've been thinking about it. It seems to me that the code will be more beautiful if I use one kind of loops in this function. What do you think?

File: lib/retry.js (hunk 1 above)
Comment (MaksGovor) [reply lib/retry.js]: Fixed, thank you.

File: lib/retry.js (hunk 2 above)
Comment (MaksGovor) [reply lib/retry.js]: Fixed, thank you.

**Pull request merged**